clevertap-core/src/main/java/com/clevertap/android/sdk/userEventLogs/UserEventLogDAOImpl.kt

```diff
@@ -0,0 +1,361 @@
+        val tName = table.tableName
+        val selection = "${Column.DEVICE_ID} = ? AND ${Column.EVENT_NAME} = ?"
+        val selectionArgs = arrayOf(deviceID, eventName)
+        return try {
+            db.readableDatabase.query(
+                tName, null, selection, selectionArgs, null, null, null, null
+            )?.use { cursor ->
+                if (cursor.moveToFirst()) {
+                    val eventLog = UserEventLog(
+                        eventName = cursor.getString(cursor.getColumnIndexOrThrow(Column.EVENT_NAME)),
+                        firstTs = cursor.getLong(cursor.getColumnIndexOrThrow(Column.FIRST_TS)),
+                        lastTs = cursor.getLong(cursor.getColumnIndexOrThrow(Column.LAST_TS)),
+                        countOfEvents = cursor.getInt(cursor.getColumnIndexOrThrow(Column.COUNT)),
+                        deviceID = cursor.getString(cursor.getColumnIndexOrThrow(Column.DEVICE_ID))
+                    )
+                    eventLog
+                } else {
+                    null
+                }
+            }
+        } catch (e: Exception) {
+            logger.verbose("Could not fetch records out of database $tName.", e)
+            null
+        }
+    }
+
+    @WorkerThread
+    override fun readEventCountByDeviceID(deviceID: String, eventName: String): Int =
+        readEventColumnByDeviceID(
+            deviceID,
+            eventName,
+            Column.COUNT,
+            defaultValueExtractor = { -1 },
+            valueExtractor = { cursor, columnName ->
+                cursor.getInt(cursor.getColumnIndexOrThrow(columnName))
+            }
+        )
+
+
+    @WorkerThread
+    override fun readEventFirstTsByDeviceID(deviceID: String, eventName: String): Long =
+        readEventColumnByDeviceID(
+            deviceID,
+            eventName,
+            Column.FIRST_TS,
+            defaultValueExtractor = { -1L },
+            valueExtractor = { cursor, columnName ->
+                cursor.getLong(cursor.getColumnIndexOrThrow(columnName))
+            }
+        )
+
+    @WorkerThread
+    override fun readEventLastTsByDeviceID(deviceID: String, eventName: String): Long =
+        readEventColumnByDeviceID(
+            deviceID,
+            eventName,
+            Column.LAST_TS,
+            defaultValueExtractor = { -1L },
+            valueExtractor = { cursor, columnName ->
+                cursor.getLong(cursor.getColumnIndexOrThrow(columnName))
+            }
+        )
+
+    @WorkerThread
+    override fun eventExistsByDeviceID(deviceID: String, eventName: String): Boolean {
+        val tName = table.tableName
+        val selection = "${Column.DEVICE_ID} = ? AND ${Column.EVENT_NAME} = ?"
+        val selectionArgs = arrayOf(deviceID, eventName)
+
+        val query = """
+            SELECT EXISTS(
+                SELECT 1 
+                FROM $tName 
+                WHERE $selection
+            ) AS eventExists;
```

Comment:
"eventExists" could be a string val, it is used in this query and in the cursor.  Same for eventExistsByDeviceIDAndCount

Reply:
fixed in https://github.com/CleverTap/clevertap-android-sdk/commit/d93ee166eb19cf70b345fa0f6d63076f5260b092